From 224cd057ff443ac3a2d56c5eafef46da5a23af1f Mon Sep 17 00:00:00 2001
From: ctang <ctang.ma@gmail.com>
Date: Fri, 12 Jun 2015 10:18:05 -0400
Subject: [PATCH 0157/1164] HIVE-10977: No need to instantiate
 MetaStoreDirectSql when HMS DirectSql is disabled
 (Chaoyu Tang, reviewed by Xuefu Zhang and Sergey
 Shelukhin)

---
 .../apache/hadoop/hive/metastore/ObjectStore.java  |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/metastore/src/java/org/apache/hadoop/hive/metastore/ObjectStore.java b/metastore/src/java/org/apache/hadoop/hive/metastore/ObjectStore.java
index 9d4f625..271f8c0 100644
--- a/metastore/src/java/org/apache/hadoop/hive/metastore/ObjectStore.java
+++ b/metastore/src/java/org/apache/hadoop/hive/metastore/ObjectStore.java
@@ -292,7 +292,9 @@ private void initialize(Properties dsProps) {
     isInitialized = pm != null;
     if (isInitialized) {
       expressionProxy = createExpressionProxy(hiveConf);
-      directSql = new MetaStoreDirectSql(pm, hiveConf);
+      if (HiveConf.getBoolVar(getConf(), ConfVars.METASTORE_TRY_DIRECT_SQL)) {
+        directSql = new MetaStoreDirectSql(pm, hiveConf);
+      }
     }
     LOG.debug("RawStore: " + this + ", with PersistenceManager: " + pm +
         " created in the thread with id: " + Thread.currentThread().getId());
-- 
1.7.9.5

